### PR TITLE
Master Volume Option

### PR DESCRIPTION
**Pull request by mercury501**
Hi there!

This PR contains the following:
Master volume level handling and storage (from @elishacloud)
Recreating the "slider" to be identical to the other existing ones
Handling input, changed settings and changes in resolution

Cheers!

### CODE REVIEW

**Comment by Polymega** [comment]
l4lWo7l

**Review by elishacloud — CHANGES_REQUESTED** [review]
Great job!  Just one comment.

File: dllmain.cpp [hunk 1]
```diff
@@ -599,6 +602,18 @@ void DelayedStart()
...
+	// Patch custom strings
+	if (EnableMasterVolume && CustomExeStrSet)
+	{
+		PatchSpeakerConfigText();
+	}
+
+	// Enable master volume slider
+	if (EnableMasterVolume)
+	{
+		PatchMasterVolumeSlider();
```
[the remark this answers is not in the document]
Comment (mercury501) [reply dllmain.cpp:609]: Sure thing @elishacloud, makes sense!

**Comment by mercury501** [comment]
Oh and te EnableMasterVolume setting would need translating in all languages other than english and italian!

**Comment by Polymega** [comment]
> Oh and the EnableMasterVolume setting would need translating in all languages other than english and italian!

@lucazandrade @IlDucci 
May I ask you two to provide translations for the following?

@lucazandrade (PTBR)
https://github.com/mercury501/Silent-Hill-2-Enhancements/blob/MasterVolumeSlider/Launcher/config_br.xml#L357-L358

@IlDucci (Spanish)
https://github.com/mercury501/Silent-Hill-2-Enhancements/blob/MasterVolumeSlider/Launcher/config_es.xml#L357-L358

**Comment by lucazandrade** [comment]
> > Oh and the EnableMasterVolume setting would need translating in all languages other than english and italian!
> 
> @lucazandrade @IlDucci May I ask you two to provide translations for the following?
> 
> @lucazandrade (PTBR) https://github.com/mercury501/Silent-Hill-2-Enhancements/blob/MasterVolumeSlider/Launcher/config_br.xml#L357-L358
> 
> @IlDucci (Spanish) https://github.com/mercury501/Silent-Hill-2-Enhancements/blob/MasterVolumeSlider/Launcher/config_es.xml#L357-L358

Done: https://github.com/elishacloud/Silent-Hill-2-Enhancements/pull/798/commits/636c2e4c7dcee6c5fc0be5886b1ceae343a68d81#diff-e7b51e253f4590bcc5cd4c5ef983ddeaf7d791e67e9dc84951eae5ccb1ad5b0f

Thanks for that, it's always a pleasure!!

**Review by elishacloud — APPROVED** [review]

**Comment by IlDucci** [comment]
Done! https://github.com/elishacloud/Silent-Hill-2-Enhancements/pull/799

**Comment by elishacloud** [comment]
Nice work!

**Comment by elishacloud** [comment]
Here is the latest merged code with the translations.  Let me know if anything is missing.

New build: [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/11763747/d3d8.zip)

**Comment by Polymega** [comment]
Thanks, Elisha! I know this is likely beyond our control, but I wanted to point out that Windows wasn't liking this test build, for some reason:

image
image

**Comment by elishacloud** [comment]
Yeah, there is no easy way to completely prevent that.  That is another False Positive.  Once the file is out for a while it will stop detecting it.

**Comment by Polymega** [comment]
@virake @IlDucci 
Looking ahead, as part of our next project update we'll be releasing a new "Master Volume" feature ([see here](https://www.youtube.com/watch?v=JEkH0E0rwUk)). This feature will utilize two new, custom text strings, which means these text strings would also need to be translated as part of your translation packs.

The text strings are:

```
Master Volume
Adjust master volume
```

At your convenience, would you please add the following lines to your respective `r_resource_*.res` files that's included in your translation pack?

image

This two new strings would be added to lines `28` and `29` in the `r_resource_*.res` file. 
_Make sure to keep the "\h" at the beginning of the text strings._

If you need to, you can grab your translation pack files from the project site: https://enhanced.townofsilenthill.com/SH2/languagepacks.htm

Thank you both

**Comment by IlDucci** [comment]
Here's Castilian Spanish:
[SH2EE_LangPack_Castilian_Spanish_V1.0.4.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/11780998/SH2EE_LangPack_Castilian_Spanish_V1.0.4.zip)

By the way, was the file name r_resource_X or r_menu_X? Was it changed?

**Comment by Polymega** [comment]
Thanks, IlDucci!

> By the way, was the file name r_resource_X or r_menu_X? Was it changed?

I wasn't on my normal computer when I wrote that, so I was going off of memory and got it incorrect. It's still `r_menu_*`. I'll update my post on that. Thanks

**Pull request merged**